### PR TITLE
Breaking specs

### PR DESCRIPTION
**Pull request by pboling**
### Proposed changes in this pull request

This PR is to show what is broken.  Some of the things broken are mentioned in the Readme as expected to be working.

#### Checklist
* [x] New tests have been added
* [ ] ~`npm test` passes successfully~
* [ ] Documentation has been updated

@amir20 to review

### CODE REVIEW

File: src/__test__/out_object.spec.js [hunk 1]
```diff
@@ -28,46 +28,187 @@ describe('Command', () => {
...
+                let data = await outObj.property('data');
+                expect(data).toEqual('test');
+            });
+        });
     });
 
-    it('#property() works with input params', async () => {
-        let page = await phantom.createPage();
-        let outObj = phantom.createOutObject();
+    describe('#property', () => {
```
Comment (pboling) [inline src/__test__/out_object.spec.js]: the Property specs here in this describe block pass.  All other specs in this change set fail.

File: src/__test__/out_object.spec.js [hunk 2]
```diff
@@ -28,46 +28,187 @@ describe('Command', () => {
...
+        'use strict';
+        describe('("onResourceRequested", true)', () => {
+            it('returns a value set by phantom', async () => {
+                let page = await phantom.createPage();
+                let outObj = phantom.createOutObject();
 
-        await page.property('onResourceReceived', function(response, out) {
-            out.lastResponse = response;
-        }, outObj);
+                await page.property('onResourceRequested', true, function(requestData, networkRequest, out) {
```
Comment (amir20) [inline src/__test__/out_object.spec.js]: @pboling why you passing `true` to `property`? That's supposed to be for `on` only according to the read me.

**Comment by amir20** [comment]
Almost all of your examples are using `page.property('onResourceReceived', true, function(response, out)` which isn't right. The README says that's only supported for `on`. Am I missing something?

**Comment by pboling** [comment]
Sorry, that was a transcription error.  Also, the specs that I care most about, because `property` is not a viable solution for my needs, are the ones with `on` that pass true.  And they are broken.  Fixing the others now.

File: src/__test__/out_object.spec.js (hunk 2 above)
Comment (pboling) [inline src/__test__/out_object.spec.js]: Oops.  This is supposed to be a test of `on`, not `property`.  Fixed in latest commit.

**Comment by pboling** [comment]
@amir20 the specs have been switched to call `on`, with `true`.

**Comment by pboling** [comment]
It is a little strange that I get 9 failures locally and Travis only gets 7, but even still all the ones testing the feature I need are failing on Travis as well as locally.

**Comment by amir20** [comment]
I see what's going on. The problem is that `on` and `createOutObject` have never been tested together. I don't think the README file suggest that it works togther, but I could see how it is confusing. I added `property` with `outobject` a long time ago and then someone else added `on`. 

Probably worth doing one of two things:
1. Update README file to say both don't work. I remember asking in the pull request what is the difference of `on` with `runOnPhantom=true` and `property` . I forget the reason. 
2. Probably deprecate on run with `runOnPhantom=true`. And fix it so that it works.

File: src/__test__/out_object.spec.js [hunk 3]
```diff
@@ -28,46 +28,187 @@ describe('Command', () => {
...
+        'use strict';
+        describe('("onResourceRequested", true)', () => {
+            it('returns a value set by phantom', async () => {
+                let page = await phantom.createPage();
+                let outObj = phantom.createOutObject();
 
-        await page.property('onResourceReceived', function(response, out) {
-            out.lastResponse = response;
-        }, outObj);
+                await page.on('onResourceRequested', true, function(requestData, networkRequest, out) {
```
Comment (amir20) [inline src/__test__/out_object.spec.js]:
```js
await page.property('onResourceRequested', function(requestData, networkRequest, out) {
                    out.lastRequest = requestData;
                }, outObj);
```
works. So probably `on` and `outobject` have not been tested. 

See my other comment

**Comment by pboling** [comment]
@amir20 the second to last line of the Readme section for `page#on` says the outObject can be used with `on`, but I agree, it looks like it does not work.

https://github.com/amir20/phantomjs-node#pageon

> The same as in property, you can pass additional params to the function in the same way, and even use the object created by #createOutObject().

**Comment by pboling** [comment]
@amir20 I have added robust working examples that accomplish what I need via `property` to this PR!

File: examples/event.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    out.isMatch = (/https?:\/\/.+?\.(:?png|gif|jpeg|jpg|svg|swf|css|woff|ttf)/gi).test(requestData.url);
+    var responseObj = {};
+    responseObj.id = requestData.id;
+    responseObj.url = requestData.url;
+    responseObj.status = requestData.status;
+    responseObj.time = requestData.time;
+    if (out.isMatch) {
+      responseObj.status = 'aborted';
+      out.urls_aborted.push(responseObj);
+      networkRequest.abort();
```
Comment (pboling) [inline examples/event.js]: It does work!!!

**Comment by amir20** [comment]
This PR is no longer valid. Closing. The original issue is still valid and documentation needs to be updated.

**Comment by pboling** [comment]
@amir20 Checking my understanding here:
1. The failing specs I added are things that are not expected to work, and the documentation just needs to be updated to reflect that.
2. The passing specs I added are good but tainted in this PR because of the presence of all the failing specs.

If that is correct I will add a new PR that corrects the documentation and also adds the additional specs.

NOTE: I was able to get my code working following the model of the passing specs!

**Comment by amir20** [comment]
> The failing specs I added are things that are not expected to work, and the documentation just needs to be updated to reflect that.

Yes. 

> The passing specs I added are good but tainted in this PR because of the presence of all the failing specs.

Also yes. I actually didn't know you had added new tests. I'd be more than happy to add those in as a separate PR if you have 'em available. I love more tests. I just couldn't pass through the broken tests.

**Comment by pboling** [comment]
I'll separate out the passing ones!

**Pull request closed without merging**